packages/vc-export/src/Presentation.ts

```diff
@@ -0,0 +1,278 @@
+  holder,
+  verifiableCredential,
+}: {
+  holder: DidUri
+  verifiableCredential: VerifiableCredential[] | VerifiableCredential
+}): void {
+  const credentials = Array.isArray(verifiableCredential)
+    ? verifiableCredential
+    : [verifiableCredential]
+  credentials.forEach(({ nonTransferable, credentialSubject, id }) => {
+    if (nonTransferable && credentialSubject.id !== holder)
+      throw new Error(
+        `The credential with id ${id} is non-transferable and cannot be presented by the identity ${holder}`
+      )
+  })
+}
+
+/**
+ * Creates a Verifiable Presentation from one or more Verifiable Credentials.
+ * This should be signed before sending to a verifier to provide authentication.
+ *
+ * @param VCs One or more Verifiable Credentials.
+ * @param holder The holder of the credentials in the presentation, which also signs the presentation.
+ * @returns An (unsigned) Verifiable Presentation containing the original VCs with its proofs.
+ */
+export function create(
+  VCs: VerifiableCredential[],
+  holder: DidUri
+): VerifiablePresentation {
+  const verifiableCredential = VCs.length === 1 ? VCs[0] : VCs
+  const presentation: VerifiablePresentation = {
+    '@context': [W3C_CREDENTIAL_CONTEXT_URL],
+    type: [W3C_PRESENTATION_TYPE],
+    verifiableCredential,
+    holder,
+  }
+  validateStructure(presentation)
+  assertHolderCanPresentCredentials(presentation)
+  return presentation
+}
+
+/**
+ * Signs a presentation in its JWT rendering.
+ *
+ * @param presentation The VerifiablePresentation (without proof).
+ * @param signingKey Key object required for signing.
+ * @param signingKey.secretKey The bytes of the secret key.
+ * @param signingKey.keyUri The key uri by which the public key can be looked up from a DID document.
+ * @param signingKey.type The key type. Ed25519 and ecdsa (secp256k1) are supported.
+ * @param options Additional optional configuration.
+ * @param options.validFrom A Date or timestamp (in ms since the epoch) indicating the earliest point in time where the presentation becomes valid.
+ * @param options.validUntil A Date or timestamp (in ms since the epoch) indicating the earliest point in time where the presentation becomes valid.
+ * @param options.expiresIn Duration of validity of the presentation in seconds. If omitted, the presentation's validity is unlimited.
+ * @param options.challenge Optional challenge provided by a verifier that can be used to prevent replay attacks.
+ * @param options.verifier Identifier of the verifier to prevent unintended re-use of the presentation.
+ * @returns A signed JWT in compact representation containing a VerifiablePresentation.
+ */
+export function signAsJwt(
+  presentation: VerifiablePresentation,
+  signingKey: {
+    secretKey: Uint8Array
+    keyUri: DidResourceUri
+    type: supportedKeys
+  },
+  options: {
+    challenge?: string
+    verifier?: string
+    validFrom?: Date | number | string
+    validUntil?: Date | number | string
```

Comment:
Yeah, I think having them in the `create` might make more sense indeed.

Reply:
gotcha, only question left for me here is the naming now. I'm debating whether to just use the names of the presentation properties this sets, which in some sense would be more straightforward - but their names are just super unintuitive, and in fact even the specs mention they shall be replaced with validFrom/validTo in the next iteration. What do you think?

Reply:
oh and whether or not expiresAt should override the expirationDate - that is, should it work only as a fallback if expirationDate is not set, or replace the expirationDate already set? I'm leaning towards the latter, because it seems like it would create less 'surprises' (i.e. expiresAt not having an effect).

Reply:
> I'm debating whether to just use the names of the presentation properties this sets

I personally think `validFrom` and `validUntil` are a better naming. We don't need to necessarily bind ourselves to the naming of the underlying types.

> oh and whether or not expiresAt should override the expirationDate - that is, should it work only as a fallback if expirationDate is not set, or replace the expirationDate already set? I'm leaning towards the latter, because it seems like it would create less 'surprises' (i.e. expiresAt not having an effect).

I agree.